src/specific/s_input.c

```diff
@@ -1,732 +1,443 @@
+        case SDL_SCANCODE_KP_AT:              return "PAD@";
+        case SDL_SCANCODE_KP_EXCLAM:          return "PAD!";
+        case SDL_SCANCODE_KP_MEMSTORE:        return "PADMS";
+        case SDL_SCANCODE_KP_MEMRECALL:       return "PADMR";
+        case SDL_SCANCODE_KP_MEMCLEAR:        return "PADMC";
+        case SDL_SCANCODE_KP_MEMADD:          return "PADMA";
+        case SDL_SCANCODE_KP_MEMSUBTRACT:     return "PADM-";
+        case SDL_SCANCODE_KP_MEMMULTIPLY:     return "PADM*";
+        case SDL_SCANCODE_KP_MEMDIVIDE:       return "PADM/";
+        case SDL_SCANCODE_KP_PLUSMINUS:       return "PAD+-";
+        case SDL_SCANCODE_KP_CLEAR:           return "PADCL";
+        case SDL_SCANCODE_KP_CLEARENTRY:      return "PADCL";
+        case SDL_SCANCODE_KP_BINARY:          return "PAD02";
+        case SDL_SCANCODE_KP_OCTAL:           return "PAD08";
+        case SDL_SCANCODE_KP_DECIMAL:         return "PAD10";
+        case SDL_SCANCODE_KP_HEXADECIMAL:     return "PAD16";
+        case SDL_SCANCODE_MODE:               return "MODE";
+        case SDL_SCANCODE_AUDIONEXT:          return "NEXT";
+        case SDL_SCANCODE_AUDIOPREV:          return "PREV";
+        case SDL_SCANCODE_AUDIOSTOP:          return "STOP";
+        case SDL_SCANCODE_AUDIOPLAY:          return "PLAY";
+        case SDL_SCANCODE_AUDIOMUTE:          return "MUTE";
+        case SDL_SCANCODE_MEDIASELECT:        return "MEDIA";
+        case SDL_SCANCODE_WWW:                return "WWW";
+        case SDL_SCANCODE_MAIL:               return "MAIL";
+        case SDL_SCANCODE_CALCULATOR:         return "CALC";
+        case SDL_SCANCODE_COMPUTER:           return "COMP";
+        case SDL_SCANCODE_AC_SEARCH:          return "SRCH";
+        case SDL_SCANCODE_AC_HOME:            return "HOME";
+        case SDL_SCANCODE_AC_BACK:            return "BACK";
+        case SDL_SCANCODE_AC_FORWARD:         return "FRWD";
+        case SDL_SCANCODE_AC_STOP:            return "STOP";
+        case SDL_SCANCODE_AC_REFRESH:         return "RFRSH";
+        case SDL_SCANCODE_AC_BOOKMARKS:       return "BKMK";
+        case SDL_SCANCODE_BRIGHTNESSDOWN:     return "BNDN";
+        case SDL_SCANCODE_BRIGHTNESSUP:       return "BNUP";
+        case SDL_SCANCODE_DISPLAYSWITCH:      return "DPSW";
+        case SDL_SCANCODE_KBDILLUMTOGGLE:     return "KBDIT";
+        case SDL_SCANCODE_KBDILLUMDOWN:       return "KBDID";
+        case SDL_SCANCODE_KBDILLUMUP:         return "KBDIU";
+        case SDL_SCANCODE_EJECT:              return "EJECT";
+        case SDL_SCANCODE_SLEEP:              return "SLEEP";
+        case SDL_SCANCODE_APP1:               return "APP1";
+        case SDL_SCANCODE_APP2:               return "APP2";
+        case SDL_SCANCODE_AUDIOREWIND:        return "RWND";
+        case SDL_SCANCODE_AUDIOFASTFORWARD:   return "FF";
     }
-    S_Shell_SpinMessageLoop();
-
-    return true;
+    // clang-format on
+    return "????";
```

Comment:
It will display to the user that they bound a key to a "????" button in the controls menu, but it will still work fine, in theory. I'm saying in theory because I covered almost all SDL defines and I can't think of a button outside of this list, maybe except the Fn key.

Reply:
Ok I figured. Makes sense.